src/components/Books.js

```diff
@@ -1,163 +1,76 @@
-              textColor: '#f88',
-              trailColor: '#d6d6d6',
-            })}
-          />
-        </div>
-        <div className="onPBtn" style={{ display: 'flex', flexDirection: 'column', paddingInlineEnd: '20%' }}>
-          <i>Current Chapter</i>
-          <button type="button" onClick={onProgressUpdate}>Update Progress</button>
-        </div>
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setBook({ ...book, [name]: value });
+  };
 
-      </div>
-    </div>
-  );
-}
+  function generateUUID() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+      const r = Math.random() * 16 | 0; const
+        v = c === 'x' ? r : (r & 0x3 | 0x8);
+      return v.toString(16);
+    });
+  }
 
-function useLocalStorage(key, initialValue) {
-  const [storedValue, setStoredValue] = React.useState(() => {
-    try {
-      const item = window.localStorage.getItem(key);
-      return item ? JSON.parse(item) : initialValue;
-    } catch (error) {
-      return initialValue;
-    }
-  });
+  const handleAddBook = async () => {
+    const newBook = {
+      item_id: generateUUID(),
+      title: book.title,
+      author: book.author,
+      category: 'Fiction',
+    };
 
-  const setValue = (value) => {
     try {
-      const valueToStore = value instanceof Function ? value(storedValue) : value;
-      setStoredValue(valueToStore);
-      window.localStorage.setItem(key, JSON.stringify(valueToStore));
+      await dispatch(addBookAsync({ app_id: selectedApp, newBook }));
     } catch (error) {
-      console.log('Error saving to local storage');
+      console.error('Error adding book:', error);
     }
   };
 
-  return [storedValue, setValue];
-}
-
-function AdditionForm() {
-  const [title, setTitle] = useState('');
-  const [author, setAuthor] = useState('');
-  const [category, setCategory] = useState('');
-  const [books, setBooks] = useLocalStorage('books', []);
-
-  const onSubmit = (e) => {
-    e.preventDefault();
-    setBooks((prevBooks) => [
-      ...prevBooks,
-      {
-        title, author, category, progress: 0,
-      },
-    ]);
-    setTitle('');
-    setAuthor('');
-    setCategory('');
-  };
-
-  const onUpdateProgressBarClick = (index) => {
-    setBooks((prevBooks) => prevBooks.map((book, i) => (i === index
-      ? { ...book, progress: Math.min(book.progress + 5, 100) }
-      : book)));
-  };
-
   return (
     <div className="Container">
       <Header />
-      <div id="dynamicDisplay">
-        {books.map((book, index) => (
-          <Book
-            key={index}
-            title={book.title}
-            author={book.author}
-            progress={book.progress}
-            category={book.category}
-            onProgressUpdate={() => onUpdateProgressBarClick(index)}
-          />
-        ))}
-      </div>
+      <BookList app_id={selectedApp} />
+      <h1 style={{ fontWeight: 'bold', color: '#888888' }}>Add Book</h1>
```

Comment:
- [ ] From here too, please remove the inline styling. I would suggest not to create your css 'inline'. It's very convenient, but doesn't reflect actual selector specificity in a real css file later, and if you keep it, you'll regret the maintenance load later.